### PR TITLE
Convert action-queue to TypeScript

### PR DESCRIPTION
**Pull request by brad-decker**
## Explanation
Converts the action-queue to typescript, which was needed to effectively type the action.js file.

## Manual Testing Steps
1. Pull branch
2. `yarn`
3. `yarn start`
4. Perform actions that hit the background such as settings toggles, etc
5. Ensure no change in behavior.

## Pre-merge author checklist

- [x] I've clearly explained:
  - [x] What problem this PR is solving
  - [x] How this problem was solved
  - [x] How reviewers can test my changes
- [ ] Sufficient automated test coverage has been added

## Pre-merge reviewer checklist

- [x] Manual testing (e.g. pull and build branch, run in browser, test code being changed)
- [ ] PR is linked to the appropriate GitHub issue
- [ ] **IF** this PR fixes a bug in the release milestone, add this PR to the release milestone

If further QA is required (e.g. new feature, complex testing steps, large refactor), add the `Extension QA Board` label.

In this case, a QA Engineer approval will be be required.

### CODE REVIEW

File: ui/store/action-queue/index.ts [hunk 1]
```diff
@@ -43,22 +57,22 @@ function failQueue() {
...
+export function dropQueue(silently: boolean) {
   if (!silently) {
     failQueue();
   }
   actionRetryQueue.length = 0;
 }
 
 // add action to queue
-const executeActionOrAddToRetryQueue = (item) => {
+const executeActionOrAddToRetryQueue = (item: BackgroundAction) => {
```
Comment (DDDDDanica) [inline ui/store/action-queue/index.ts:70]: Can we add a return type here?

File: ui/store/action-queue/index.ts [hunk 2]
```diff
@@ -43,22 +57,22 @@ function failQueue() {
  * Drops the entire actions queue. Rejects all actions in the queue unless silently==true
  * Does not affect the single action that is currently being processed.
  *
- * @param {boolean} [silently]
+ * @param [silently]
  */
-export function dropQueue(silently) {
+export function dropQueue(silently: boolean) {
```
Comment (DDDDDanica) [inline ui/store/action-queue/index.ts:62]: Can we add a return type here?

File: ui/store/action-queue/index.ts (hunk 1 above)
Comment (brad-decker) [reply ui/store/action-queue/index.ts]: Done

File: ui/store/action-queue/index.ts (hunk 2 above)
Comment (brad-decker) [reply ui/store/action-queue/index.ts]: Done :D

**Review by DDDDDanica — APPROVED** [review]

**Review by darkwing — APPROVED** [review]

File: ui/store/action-queue/index.ts [hunk 3]
```diff
@@ -24,10 +23,25 @@ import { isManifestV3 } from '../../../shared/modules/mv3.utils';
 //   }, {});
 // }
 
-let background = null;
-let promisifiedBackground = null;
+let background:
+  | ({
```
Comment (pedronfigueiredo) [inline ui/store/action-queue/index.ts:27]: Does it make sense to create variables for these types or they just aren't used anywhere else?

Comment (brad-decker) [reply ui/store/action-queue/index.ts:27]: It may make sense if we find them used elsewhere but as of right now they aren't.

**Pull request merged**